backend/main.py

```
@@ -63,32 +64,70 @@
 @asynccontextmanager
 async def combined_lifespan(app):
     """Combined lifespan manager for OAuth, session, and Redis cleanup tasks"""
-    # Validate environment on startup
-    _validate_environment()
+    # Validate environment on startup (non-blocking - log warnings instead of crashing)
+    try:
+        _validate_environment()
+        logger.info("✅ Environment validation passed")
+    except RuntimeError as e:
+        logger.error(f"⚠️  Environment validation failed: {e}")
```

Ruff (0.14.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)